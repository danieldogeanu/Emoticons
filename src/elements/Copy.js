import React from 'react';

function Copy(props) {
	const hint = 'Copy the emoticon and paste it where you need it.';
	return (
		<span className="Copy" title={hint}>
			{props.children}
		</span>
	);
}

export default Copy;
