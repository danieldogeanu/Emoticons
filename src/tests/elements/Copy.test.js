import React from 'react';
import {render} from '@testing-library/react';
import Copy from '../../elements/Copy';

describe('Copy Element', () => {

	it('renders copy element properly', () => {
		const btnText = 'Copy Me';
		const {getByTitle, getByText} = render(<Copy><button>{btnText}</button></Copy>);
		const renderedCopy = getByTitle(/copy the emoticon/i);
		const renderedBtn = getByText(btnText);

		expect(renderedCopy).toBeInTheDocument();
		expect(renderedCopy).toHaveTextContent(btnText);
		expect(renderedCopy).toHaveClass('Copy');
		expect(renderedCopy).toContainElement(renderedBtn);
		expect(renderedCopy).toMatchSnapshot();
	});

});
