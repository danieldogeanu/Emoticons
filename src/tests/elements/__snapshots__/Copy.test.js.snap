// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy Element renders copy element properly 1`] = `
<span
  class="Copy"
  title="Copy the emoticon and paste it where you need it."
>
  <button>
    Copy Me
  </button>
</span>
`;
